lib/collection/src/operations/universal_query/planned_query.rs

```diff
@@ -72,10 +74,16 @@ pub struct MergePlan {
     pub rescore_params: Option<RescoreParams>,
 }
 
-impl TryFrom<ShardQueryRequest> for PlannedQuery {
-    type Error = CollectionError;
+impl PlannedQuery {
+    pub fn with_capacity(capacity: usize) -> Self {
+        Self {
+            root_plans: Vec::with_capacity(capacity),
+            searches: Vec::with_capacity(capacity),
+            scrolls: Vec::with_capacity(capacity),
+        }
+    }
```

Comment:
It is unlikely that all 3 are going to have same capacity, isn't it?

Reply:
I propose to just init empty, don't think it is critical here

Reply:
🤷 micro-optimization, I am fine either way